JSON-connect/src/main.js

```diff
@@ -159,51 +177,144 @@ function fetchData(url, cache) {
   if (!url || !url.match(/^https?:\/\/.+$/g)) {
     sendUserError('"' + url + '" is not a valid url.');
   }
-
-  var content = cache ? getCachedData(url) : fetchJSON(url);
-
+  try {
+    var content = cache ? getCachedData(url) : fetchJSON(url);
+  } catch (e) {
+    sendUserError(
+      'Unable to download or cache "' +
+        url +
+        '". Error: \n' +
+        e +
+        '\n' +
+        e.stack
```

Comment:
We don't want to show stacktraces to regular users. Try to stick to an error message that makes sense to everyone.

Reply:
makes sense, done.